pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -91,7 +92,20 @@ func (a *v1beta1PolicyApplier) setAuthnFilterForPeerAuthn(proxyType model.NodeTy
 	p := config.Policy
 	p.Peers = []*authn_alpha.PeerAuthenticationMethod{}
 
-	effectiveMTLSMode := a.getMutualTLSModeForPort(port)
+	var effectiveMTLSMode model.MutualTLSMode
+	if proxyType == model.SidecarProxy {
+		effectiveMTLSMode = a.getMutualTLSModeForPort(port)
+	} else {
+		// this is for gateway with a server whose TLS mode is ISTIO_MUTUAL
+		// this is effectively the same as strict mode. We dont really
+		// care about permissive or strict here. We simply need to validate that the peer cert is
+		// a proper spiffe cert so that authz policies can use source principal based validations here.
+		effectiveMTLSMode = model.MTLSStrict
+		// we should accept traffic from any trust domain. We expect the use of authZ policies to
```

Comment:
So for sidecar mTLS, we implicitly validate trust domain, etc but for Gateway we only do so based on policies? Is that clearly documented? Its not immediately obvious to me that ISTIO_MUTUAL on gw would behave different than sidecar

Reply:
In multicluster scenario, if the trust domain is validated, there is no way for the user to allow traffic from other clusters in different trust domains. Hence the rationale to not validate

Reply:
Isn't this MUTUAL then, not ISTIO_MUTUAL? Not sure how its "Istio" mutual if its not the same rtrust domain?

In general, should we add authn filter for all mutual not just istio mutual?